sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -340,13 +341,15 @@ private[state] class HDFSBackedStateStoreProvider(
 
   private def updateFromDeltaFile(version: Long, map: MapType): Unit = {
     val fileToRead = deltaFile(version)
-    if (!fs.exists(fileToRead)) {
-      throw new IllegalStateException(
-        s"Error reading delta file $fileToRead of $this: $fileToRead does not exist")
-    }
     var input: DataInputStream = null
     try {
-      input = decompressStream(fs.open(fileToRead))
+      try {
```

Comment:
I feel like we can get rid of the nesting of the try here by assigning input to the result of the inner try {} you added. If it succeeds, then it opens a second try block that ensure the (non-null) input stream is closed.